packages/core/src/hydration/utils.ts

```diff
@@ -123,6 +146,72 @@ export function getComponentLViewForHydration(viewRef: ViewRef): LView|null {
   return lView;
 }
 
+/**
+ * Restores text nodes and separators into the DOM that were lost during SSR
+ * serialization. The hydration process replaces empty text nodes and text
+ * nodes that are immediately adjacent to other text nodes with comment nodes
+ * that this method filters on to restore those missing nodes that the
+ * hydration process is expecting to be present.
+ *
+ * @param node The app's root HTML Element
+ */
+export function processTextNodeMarkersBeforeHydration(node: HTMLElement) {
+  const doc = getDocument();
+  const commentIterator = doc.createNodeIterator(node, NodeFilter.SHOW_COMMENT, {
+    acceptNode(node) {
+      const content = node.textContent;
```

Comment:
I'd propose doing the trimming here as well (similar to what we do during serialization, may be extracting this into a separate fn, like `getTextNodeContent`):
```suggestion
      const content = node.textContent.replace(/\s/gm, '');
```